### PR TITLE
fix: map search shows suggestion for coordinates

### PR DESCRIPTION
**Pull request by knipec**
## Description
The commits should be pretty reasonable to look through if you want to review that way. 

### Changes
- Typing valid lat, long into map search should now show a suggestion for those coordinates instead of zooming the map directly there and opening a callout. This makes the debounce unnecessary.
- Should be able to zoom to a lat, long when offline
- Aside: latitude +/- 90 should be valid for coordinates
- Restructure to reduce complexity
- Add some tests

### Related Issues
Fixes https://github.com/techmatters/terraso-mobile-client/issues/3200 
By implementing https://github.com/techmatters/terraso-mobile-client/issues/2017 

### Verification steps
- Type a valid lat,long in map search
- Confirm clicking the suggestion moves you to that lat,long
- Confirm that regular mapbox suggestions also continue to work
- Confirm search still disappears when the callout is open

### CODE REVIEW

File: dev-client/src/util.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export const sortCompare = (
 
 export const isValidCoordinates = (input: string) => {
   const coordRegex =
-    /^([-+]?[1-8]?\d(?:\.\d+)?),\s*([-+]?180(?:\.0+)?|[-+]?((1[0-7]\d)|([1-9]?\d))(?:\.\d+)?)$/;
+    /^([-+]?90(?:\.0+)?|[-+]?[1-8]?\d(?:\.\d+)?),\s*([-+]?180(?:\.0+)?|[-+]?((1[0-7]\d)|([1-9]?\d))(?:\.\d+)?)$/;
```
Comment (johannesparty) [inline dev-client/src/util.ts]: I like using regex for this, but hard to read in this form.  might be worth adding a comment explaining what this captures.  Also just curious what you used to write the regex.  I used to use online regex tools and hand-edit and check with examples you could put into the tool.  But now ... claude?

File: dev-client/src/util.test.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      expect(isValidCoordinates('12.345,67.890')).toBe(true);
+    });
+
+    it('accepts negative coordinates', () => {
+      expect(isValidCoordinates('-12.345, -67.890')).toBe(true);
+      expect(isValidCoordinates('-12.345, 67.890')).toBe(true);
+      expect(isValidCoordinates('12.345, -67.890')).toBe(true);
+    });
+
+    it('accepts boundary values', () => {
```
Comment (johannesparty) [inline dev-client/src/util.test.ts]: Looks like these are the cases where you fixed the regex :)
This is almost documentation.  But could add the summary of what should/shouldn't work here too I guess.

**Review by johannesparty — APPROVED** [review]
Looks good, lots of tests too :).
A couple of behaviors:
- in simulator, if you type coordinate and hit enter, nothing happens, you have to click on the lat,long suggestion.  not important because doesn't really matter for real phones unless some is using an external keyboard (rare?).
- I think the algorithm adds lat,long suggestion if the input has the syntax for it (regex), and then adds the mapbox suggestions.  when given coordinates, do we ever get mapbox suggestions?  if not, we could (a) skip 2nd part, and (b) make it so any num,num is handled specially, but shows an error saying must be within xyz range.  again, not a problem, just a thought.
- I'm not sure it is obvious you can type in coordinates, seems like a help text below the input box would be nice, but I guess that's more a question for Courtney.
